content/operate/oss_and_stack/stack-with-enterprise/release-notes/redistimeseries/redistimeseries-1.12-release-notes.md

```diff
@@ -0,0 +1,53 @@
+---
+Title: RedisTimeSeries 1.12 release notes
+alwaysopen: false
+categories:
+- docs
+- operate
+- stack
+description: RedisTimeSeries 1.12 adds a highly requested feature - insertion-filter for close samples.
+linkTitle: v1.12 (July 2024)
+min-version-db: '7.4'
+min-version-rs: 7.6 (TBD)
+toc: 'true'
+weight: 94
+---
+## Requirements
+
+RedisTimeSeries v1.12.2 requires:
+
+- Minimum Redis compatibility version (database): 7.4
+- Minimum Redis Enterprise Software version (cluster): 7.6 (TBD)
+
+## v1.12 GA (v1.12.2) (July 2024)
+
+This is the General Availability release of RedisTimeSeries 1.12
+
+### Headlines
+
+RedisTimeSeries 1.12 adds a highly requested feature: insertion-filter for close samples. Many sensors report data periodically. Often, the difference between the measured value and the previous measured value is negligible and related to random noise or to measurement accuracy limitations. When both the time difference and the value difference between the current and the previous sample are small, it may be preferable to ignore (not to add) the new measurement.
+
+### What's new in 1.12
+
+- [#1543](https://github.com/RedisTimeSeries/RedisTimeSeries/pull/1543) insertion-filter for close samples:
+
+  - Two new [module configuration parameters](https://redis.io/docs/data-types/timeseries/configuration/) are introduced: `IGNORE_MAX_TIME_DIFF` and `IGNORE_MAX_VAL_DIFF`.
+  - Two new similar per-key parameters are introduced: `ignoreMaxTimeDiff` and `ignoreMaxValDiff`.
+  - `TS.ADD`, `TS.INCRBY`, and `TS.DECRBY` now have a new optional argument: `[IGNORE ignoreMaxTimeDiff ignoreMaxValDiff]`.
+    When creating a new time series, these two values are used to set the per-key parameters and override the two module configuration parameters. These values are ignored when specified with an existing time series.
+  - `[IGNORE ignoreMaxTimeDiff ignoreMaxValDiff]` is also supported by `TS.ALTER`.
+  - For each call to `TS.ADD`, if the following condition is met: (series is not a compaction), (the series' `DUPLICATE_POLICY` is `LAST`), (`timestamp` ≥ `max_timestamp`), (`timestamp` - `max_timestamp` ≤ `ignoreMaxTimeDiff`), and abs(`value` - `value_at_max_timestamp`) ≤ `ignoreMaxValDiff`)) then this sample is ignored (not added) to the time series. `max_timestamp` is the maximal timestamp in the time series.
```

Comment:
There might be better ways to format this but it's quite dense and hard to read as it is.
```suggestion
  - For each call to `TS.ADD`, if the following conditions are met:
      - series is not a compaction
      - the series' `DUPLICATE_POLICY` is `LAST`
      - `timestamp` ≥ `max_timestamp`
      - `timestamp` - `max_timestamp` ≤ `ignoreMaxTimeDiff`
      - abs(`value` - `value_at_max_timestamp`) ≤ `ignoreMaxValDiff`))
    then this sample is ignored (not added) to the time series. `max_timestamp` is the maximal timestamp in the time series.
```